### PR TITLE
[SPARK-37480][K8S][DOC] Sync Kubernetes configuration to latest in running-on-k8s.md

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Sync Kubernetes configurations to latest in doc

### Why are the changes needed?
Configurations in docs/running-on-kubernetes.md are not uptodate

### Does this PR introduce _any_ user-facing change?
No, docs only

### How was this patch tested?
CI passed

### CODE REVIEW

**Comment by Yikun** [comment]
cc @dongjoon-hyun @holdenk

File: docs/running-on-kubernetes.md [hunk 1]
```diff
@@ -591,7 +591,7 @@ See the [configuration page](configuration.html) for information on Spark config
   <td><code>spark.kubernetes.container.image.pullPolicy</code></td>
   <td><code>IfNotPresent</code></td>
   <td>
-    Container image pull policy used when pulling images within Kubernetes.
+    Container image pull policy used when pulling images within Kubernetes. Valid values are Always, Never, and IfNotPresent.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:594]: Could you codify the values? For example, `<code>Always</code>`?

File: docs/running-on-kubernetes.md [hunk 2]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.configMap.maxSize</code></td>
+  <td>1572864</td>
+  <td>
+    Max size limit for a config map. This is configurable as per https://etcd.io/docs/v3.4.0/dev-guide/limit/ on k8s server end.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1354]: Could you make it as a hyper link by using `<a href=...>`?

File: docs/running-on-kubernetes.md [hunk 3]
```diff
@@ -1278,6 +1295,14 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.dynamicAllocation.deleteGracePeriod</code></td>
+  <td>5s</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1300]: I guess we need `<code>` tag around this.

File: docs/running-on-kubernetes.md [hunk 4]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
   <td>3.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.configMap.maxSize</code></td>
+  <td>1572864</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1352]: Codify?

File: docs/running-on-kubernetes.md [hunk 5]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    Max size limit for a config map. This is configurable as per https://etcd.io/docs/v3.4.0/dev-guide/limit/ on k8s server end.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.allocation.executor.timeout</code></td>
+  <td>600s</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1360]: Codify?

File: docs/running-on-kubernetes.md [hunk 6]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    Time to wait before a newly created executor POD request, which does not reached
+    the POD pending state yet, considered timedout and will be deleted.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.executor.missingPodDetectDelta</code></td>
+  <td>30s</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1369]: ditto.

File: docs/running-on-kubernetes.md [hunk 7]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    When a registered executor's POD is missing from the Kubernetes API server's polled
+    list of PODs then this delta time is taken as the accepted time difference between the
+    registration time and the time of the polling. After this time the POD is considered
+    missing from the cluster and the executor will be removed.
+  </td>
+  <td>3.1.1</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.allocation.driver.readinessTimeout</code></td>
+  <td>1s</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1380]: ditto

File: docs/running-on-kubernetes.md [hunk 8]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    Time to wait for driver pod to get ready before creating executor pods. This wait
+    only happens on application start. If timeout happens, executor pods will still be
+    created.
+  </td>
+  <td>3.1.3</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.decommission.script</code></td>
+  <td>/opt/decom.sh</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1390]: ditto.

File: docs/running-on-kubernetes.md [hunk 9]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    The location of the script to use for graceful decommissioning.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.service.deleteOnTermination</code></td>
+  <td>true</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1398]: ditto

File: docs/running-on-kubernetes.md [hunk 10]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    If true, driver service will be deleted on Spark application termination. If false, it will be cleaned up when the driver pod is deletion.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.ownPersistentVolumeClaim</code></td>
+  <td>false</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1406]: ditto

File: docs/running-on-kubernetes.md [hunk 11]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    including persistent volume claims are not reusable yet. Spark will create new
+    persistent volume claims when there exists no reusable one. In other words, the total
+    number of persistent volume claims can be larger than the number of running executors
+    sometimes. This config requires <code>spark.kubernetes.driver.ownPersistentVolumeClaim=true.</code>
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.executor.disableConfigMap</code></td>
+  <td>false</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1429]: ditto

File: docs/running-on-kubernetes.md [hunk 12]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    If true, disable ConfigMap creation for executors.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.driver.pod.featureSteps</code></td>
+  <td>(none)</td>
+  <td>
+    Class names of an extra driver pod feature step implementing
+    KubernetesFeatureConfigStep. This is a developer API. Comma separated.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1440]: Codify with `<code>KubernetesFeatureConfigStep</code>` or back-quotation?

File: docs/running-on-kubernetes.md [hunk 13]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    Runs after all of Spark internal feature steps."
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.executor.pod.featureSteps</code></td>
+  <td>(none)</td>
+  <td>
+    Class name of an extra executor pod feature step implementing
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1449]: `Class name` -> `Class names`?

File: docs/running-on-kubernetes.md [hunk 14]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    application. Those newly requested executors which are unknown by Kubernetes yet are
+    also counted into this limit as they will change into pending PODs by time.
+    This limit is independent from the resource profiles as it limits the sum of all
+    allocation for all the used resource profiles.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.allocation.pods.allocator</code></td>
+  <td>direct</td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1469]: codify?

File: docs/running-on-kubernetes.md [hunk 15]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>
+    Allocator to use for pods. Possible values are direct (the default) and statefulset,
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1471]: codify?

File: docs/running-on-kubernetes.md [hunk 16]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+    or a full class name of a class implementing AbstractPodsAllocator.
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1472]: Codify `AbstractPodsAllocator`?

File: docs/running-on-kubernetes.md [hunk 17]
```diff
@@ -1322,6 +1347,144 @@ See the [configuration page](configuration.html) for information on Spark config
...
+  <td>(none)</td>
+  <td>
+    Class name of an extra executor pod feature step implementing
+    KubernetesFeatureConfigStep. This is a developer API. Comma separated.
+    Runs after all of Spark internal feature steps.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.allocation.maxPendingPods</code></td>
```
Comment (dongjoon-hyun) [inline docs/running-on-kubernetes.md:1456]: Is there a reason why you put `spark.kubernetes.allocation` after `spark.kubernetes.executor`?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you, @Yikun . I added some comments.

**Comment by Yikun** [comment]
@dongjoon-hyun Thanks for your review! codify problme and nits have been addressed!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for improving many areas including this, @Yikun .

**Comment by dongjoon-hyun** [comment]
Merged to master.

If you don't mind, could you make a backporting PR to branch-3.2 after excluding `3.3.0` configs?

**Comment by Yikun** [comment]
> If you don't mind, could you make a backporting PR to branch-3.2 after excluding 3.3.0 configs?

@dongjoon-hyun Sure, will backport it soon. Thanks for review!

**Pull request closed without merging**